### PR TITLE
IGNITE-17577 Timestamp replaced with HybridTimestamp in MvPartitionStorage.

### PR DESCRIPTION
**Pull request by ibessonov**
https://issues.apache.org/jira/browse/IGNITE-17577

### CODE REVIEW

File: modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/Timestamps.java [hunk 1]
```diff
@@ -17,62 +17,72 @@
...
-        long nodeId = getLong(pageAddr, offset);
-        long localTimestamp = getLong(pageAddr, offset + Long.BYTES);
+    static @Nullable HybridTimestamp readTimestamp(long pageAddr, int offset) {
+        long physical = getLong(pageAddr, offset);
+        int logical = getInt(pageAddr, offset + Long.BYTES);
 
-        Timestamp timestamp = new Timestamp(localTimestamp, nodeId);
-        if (timestamp.equals(RowVersion.NULL_TIMESTAMP)) {
-            timestamp = null;
+        if (physical == 0L && logical == 0) {
```
Comment (rpuch) [inline modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/Timestamps.java:44]: I suggest to extract these zeroes as constants and use them in writing methods as well. Just a little more connectivity in the code seems to be a good thing.

File: modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/RowVersion.java [hunk 2]
```diff
@@ -17,41 +17,39 @@
...
     /** Represents an absent partitionless link. */
     public static final long NULL_LINK = 0;
 
-    private static final int TIMESTAMP_STORE_SIZE_BYTES = 2 * Long.BYTES;
     private static final int NEXT_LINK_STORE_SIZE_BYTES = PartitionlessLinks.PARTITIONLESS_LINK_SIZE_BYTES;
     private static final int VALUE_SIZE_STORE_SIZE_BYTES = Integer.BYTES;
 
     public static final int TIMESTAMP_OFFSET = 0;
-    public static final int NEXT_LINK_OFFSET = TIMESTAMP_STORE_SIZE_BYTES;
+    public static final int NEXT_LINK_OFFSET = HYBRID_TIMESTAMP_SIZE;
```
[the remark this answers is not in the document]
Comment (ibessonov) [reply modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/RowVersion.java:44]: Makes sense, thank you!

File: modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/Timestamps.java [hunk 3]
```diff
@@ -17,62 +17,72 @@
...
 import java.nio.ByteBuffer;
-import org.apache.ignite.internal.tx.Timestamp;
+import org.apache.ignite.hlc.HybridTimestamp;
 import org.jetbrains.annotations.Nullable;
 
 /**
- * Code to work with {@link Timestamp}s.
+ * Code to work with {@link HybridTimestamp}s.
  */
 public class Timestamps {
```
[the remark this answers is not in the document]
Comment (ibessonov) [reply modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/Timestamps.java:33]: I'll rename it

File: modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/Timestamps.java (hunk 1 above)
Comment (ibessonov) [reply modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/Timestamps.java:44]: Ok, although I don't think it's necessary. Not every 0 deserve a constant :)

**Pull request merged**